fastscanner/adapters/realtime/polygon_realtime.py

```diff
@@ -0,0 +1,166 @@
+        )
+        if self._client is None:
+            raise RuntimeError(
+                "WebSocketClient initialization failed — client is None."
+            )
+
+        self._running = True
+        logger.info("Connecting WebSocket")
+
+        self._ws_task = asyncio.create_task(self._client.connect(self.handle_messages))
+
+    async def stop(self):
+        if not self._running:
+            logger.warning("WebSocket is not running.")
+            return
+        if self._client is None:
+            raise RuntimeError("WebSocketClient is None during stop.")
+
+        if self._symbols:
+            await self.unsubscribe(self._symbols)
+
+        try:
+            await self._client.close()
+            if self._ws_task:
+                await self._ws_task
+        except ConnectionClosedError as e:
+            logger.warning(f"Client was disconnected: {e}")
+        finally:
+            self._running = False
+            logger.info("WebSocket stopped.")
+
+    async def subscribe(self, symbols: set[str]):
+        if not self._running:
+            logger.warning("WebSocket is not running")
+            return
+        if self._client is None:
+            raise RuntimeError("WebSocketClient is None during subscribe.")
+
+        if not symbols:
+            logger.warning("No symbols to subscribe.")
+            return
+
+        tickers = [f"AM.{symbol}" for symbol in symbols]
+        self._client.subscribe(*tickers)
+        self._symbols.update(symbols)
+        logger.info(f"Subscribed to: {tickers}")
+
+    async def unsubscribe(self, symbols: set[str]):
+        if not self._running:
+            logger.warning("WebSocket is not running.")
+            return
+
+        if self._client is None:
+            raise RuntimeError("WebSocketClient is None during unsubscribe.")
+
+        if not symbols:
+            logger.warning("No symbols to unsubscribe.")
+            return
+
+        tickers = [f"AM.{symbol}" for symbol in symbols]
+        try:
+            self._client.unsubscribe(*tickers)
+            self._symbols.difference_update(symbols)
+            logger.info(f"Unsubscribed from: {tickers}")
+        except ConnectionClosedError as e:
+            logger.warning(f"WebSocket connection was already closed: {e}")
+
+    async def handle_messages(self, msgs: list[WebSocketMessage]):
+        logger.info(f"Received messages: {msgs}")
+        data = []
+        for msg in msgs:
+            record = {
+                "symbol": getattr(msg, "symbol", None),
+                "timestamp": getattr(msg, "start_timestamp", None),
+                "open": getattr(msg, "open", None),
+                "high": getattr(msg, "high", None),
+                "low": getattr(msg, "low", None),
+                "close": getattr(msg, "close", None),
+                "volume": getattr(msg, "volume", None),
+            }
+            data.append(record)
+
+        if len(data) > 0:
+            df = pd.DataFrame(data)
```

Comment:
There is no need to transform into a dataframe. This is probably reducing the performance because you're deserializing the json into an object and then serializing it back.